### PR TITLE
Minor fixes to improve django-stubs

### PR DESCRIPTION
**Pull request by PIG208**
## Related issues
None. This batch of fixes are found the project of fixing 2k+ errors in the django-stubs migration of Zulip (zulip/zulip#18777). As they are mostly minor fixes, I integrate them in this PR.

### CODE REVIEW

File: django-stubs/contrib/staticfiles/views.pyi [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from typing import Any
 
-from django.core.handlers.wsgi import WSGIRequest
+from django.http.request import HttpRequest
 from django.http.response import FileResponse
 
-def serve(request: WSGIRequest, path: str, insecure: bool = ..., **kwargs: Any) -> FileResponse: ...
+def serve(request: HttpRequest, path: str, insecure: bool = ..., **kwargs: Any) -> FileResponse: ...
```
Comment (sobolevn) [inline django-stubs/contrib/staticfiles/views.pyi:6]: 👍

File: django-stubs/contrib/auth/backends.pyi [hunk 2]
```diff
@@ -10,9 +10,7 @@ _AnyUser = Union[Model, AnonymousUser]
 UserModel: Any
 
 class BaseBackend:
-    def authenticate(
-        self, request: HttpRequest, username: Optional[str] = ..., password: Optional[str] = ..., **kwargs: Any
-    ) -> Optional[AbstractBaseUser]: ...
+    def authenticate(self, request: Optional[HttpRequest], **kwargs: Any) -> Optional[AbstractBaseUser]: ...
```
Comment (sobolevn) [inline django-stubs/contrib/auth/backends.pyi:13]: Yeap, current `3.2.x` django agrees with this signature: https://github.com/django/django/blob/3.2.6/django/contrib/auth/backends.py#L9

File: django-stubs/core/mail/backends/base.pyi [hunk 3]
```diff
@@ -13,4 +13,4 @@ class BaseEmailBackend:
     def __exit__(
         self, exc_type: Type[BaseException], exc_value: BaseException, traceback: types.TracebackType
     ) -> None: ...
-    def send_messages(self, email_messages: Iterable[EmailMessage]) -> int: ...
+    def send_messages(self, email_messages: List[EmailMessage]) -> int: ...
```
Comment (sobolevn) [inline django-stubs/core/mail/backends/base.pyi]: Can you please explain this change? Why `Iterable` is not good enough?

Comment (sobolevn) [reply django-stubs/core/mail/backends/base.pyi]: Moreover, `List` is invariant. It might cause problems.

**Review by sobolevn — COMMENTED** [review]
Thanks a lot! Great to hear that `zulip` is going to use `django-stubs` 🎉

File: django-stubs/core/mail/backends/base.pyi (hunk 3 above)
Comment (PIG208) [reply django-stubs/core/mail/backends/base.pyi]: Given the way `send_messages` is invoked (https://github.com/django/django/blob/3.2.6/django/core/mail/__init__.py#L83-L87), it is more appropriate to type it as list-like object. This allows the implementation of subclasses to use methods like `__len__` and `__getitem__`, and overwritting the method with signature of `def send_messages(self, email_messages: Iterable[EmailMessage]) -> int` will still be acceptable.

Comment (PIG208) [reply django-stubs/core/mail/backends/base.pyi]: Since `List` is invariant, perhaps `Sequence` will be better here. The PR has been updated.

**Review by sobolevn — APPROVED** [review]
Thanks a lot! 👍

File: django-stubs/contrib/staticfiles/views.pyi (hunk 1 above)
Comment (sobolevn) [reply django-stubs/contrib/staticfiles/views.pyi:6]: Btw, if you have time/will you can refactor almost all `WSGIRequest` references into `HttpRequest`

**Pull request merged**